yt_dlp/extractor/zingmp3.py

```diff
@@ -35,61 +60,82 @@ def _extract_item(self, item, fatal):
+                    })
         if not formats:
             if not fatal:
                 return
-            msg = item['msg']
+            msg = item.get('msg')
             if msg == 'Sorry, this content is not available in your country.':
                 self.raise_geo_restricted(countries=self._GEO_COUNTRIES, metadata_available=True)
             self.raise_no_formats(msg, expected=True)
         self._sort_formats(formats)
 
-        subtitles = None
         lyric = item.get('lyric')
-        if lyric:
-            subtitles = {
-                'origin': [{
-                    'url': lyric,
-                }],
-            }
+        if not lyric:
+            api = self.get_api_with_signature(name_api=self._SLUG_API.get("lyric"), param={'id': item_id})
+            info_lyric = self._download_json(api, video_id=item_id)
+            lyric = traverse_obj(info_lyric, ('data', 'file'))
+        subtitles = {
+            'origin': [{
+                'url': lyric,
+            }],
+        } if lyric else None
 
         album = item.get('album') or {}
 
         return {
             'id': item_id,
             'title': title,
             'formats': formats,
-            'thumbnail': item.get('thumbnail'),
+            'thumbnail': try_get(item, lambda x: x['thumbnail']) or try_get(item, lambda x: x['thumbnailM']),
             'subtitles': subtitles,
-            'duration': int_or_none(item.get('duration')),
+            'duration': int_or_none(try_get(item, lambda x: x['duration'])),
             'track': title,
-            'artist': item.get('artists_names'),
-            'album': album.get('name') or album.get('title'),
-            'album_artist': album.get('artists_names'),
+            'artist': try_get(item, lambda x: x['artistsNames']) or try_get(item, lambda x: x['artists_names']),
+            'album': try_get(album, lambda x: x['name']) or try_get(album, lambda x: x['title']),
+            'album_artist': try_get(album, lambda x: x['artistsNames']) or try_get(album, lambda x: x['artists_names']),
         }
 
     def _real_extract(self, url):
-        page_id = self._match_id(url)
-        webpage = self._download_webpage(
-            url.replace('://zingmp3.vn/', '://mp3.zing.vn/'),
-            page_id, query={'play_song': 1})
-        data_path = self._search_regex(
-            r'data-xml="([^"]+)', webpage, 'data path')
-        return self._process_data(self._download_json(
-            'https://mp3.zing.vn/xhr' + data_path, page_id)['data'])
+        song_id, type_url = self._match_valid_url(url).group('id', 'type')
+
+        api = self.get_api_with_signature(name_api=self._SLUG_API[type_url], param={'id': song_id})
+
+        # Check the cookie file, if cookie file doesn't exist, create a dummy request to update the cookie
+        if not self._downloader.params.get('cookiefile'):
+            self._download_json(api, video_id=song_id, note='Dummy request to update cookie')
```

Comment:
if user does `yt-dlp URL1 URL2`, this should only happen once, right? In that case, move it to `_real_initialize`.

Also, the check should be `not self.get_param('cookiefile') and not self.get_param('cookiesfrombrowser')`